app/Http/Controllers/Api/AssetsController.php

```diff
@@ -49,10 +51,32 @@ class AssetsController extends Controller
      * @since [v4.0]
      * @return JsonResponse
      */
-    public function index(Request $request, $audit = null)
+    public function index(Request $request, $audit = null) 
     {
 
-        $this->authorize('index', Asset::class);
+        \Log::debug(Route::currentRouteName());
+        
+
+        /**
+         * This looks MAD janky (and it is), but the AssetsController@index does a LOT of heavy lifting throughout the 
+         * app. This bit here just makes sure that someone without permission to view assets doesn't 
+         * end up with priv escalations because they asked for a different endpoint. 
+         * 
+         * Since we never gave the specification for which transformer to use before, it should default 
+         * gracefully to just use the AssetTransformer by default, which shouldn't break anything. 
+         * 
+         * It was either this mess, or repeating ALL of the searching and sorting and filtering code, 
+         * which would have been far worse of a mess. *sad face*  - snipe (Sept 1, 2021)
+         */
+        if (Route::currentRouteName()=='api.depreciation-report.index') {
+            $transformer = 'App\Http\Transformers\DepreciationReportTransformer';
```

Comment:
I *think* you can namespace this class in and use instead of:

`'App\Http\Transformers\DepreciationReportTransformer'`

using something along the lines of this:

```
use App\Http\Transformers;
/* ... */
DepreciationReportTransformer::class
```

This is the direction that Laravel is going - using *real* class "objects" rather than just simple strings. It means that if we have a typo, or we mess up our namespaces, the code will literally not compile - loading a page will fail. That's probably what you want (and me, to a lesser extent).

Reply:
I tried that and it didn't work - but remember, this is on dev/master, not Laravel v8